example/spec/distributionAPI/distributionSpec.js

```diff
@@ -76,51 +97,109 @@ describe('Distribution API', () => {
       );
+      privateFileUrl = getDistributionFileUrl({
+        bucket: privateBucketName, key: fileKey
+      });
+      publicFileUrl = getDistributionFileUrl({
+        bucket: publicBucketName, key: fileKey
+      });
     });
 
     afterAll(async () => {
       await accessTokensModel.delete({ accessToken });
     });
 
-    it('redirects to Earthdata login for unauthorized requests', async () => {
-      const response = await got(
-        fileUrl,
-        { followRedirect: false }
-      );
-      const authorizeUrl = new URL(response.headers.location);
-      expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
-      expect(authorizeUrl.searchParams.get('state')).toEqual(`/${config.bucket}/${fileKey}`);
-      expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
-    });
+    describe('an unauthorized user', () => {
+      it('redirects to Earthdata login for unauthorized requests', async () => {
+        const response = await got(
+          fileUrl,
+          { followRedirect: false }
+        );
+        const authorizeUrl = new URL(response.headers.location);
+        expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
+        expect(authorizeUrl.searchParams.get('state')).toEqual(`/${protectedBucketName}/${fileKey}`);
+        expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+      });
 
-    it('redirecting to Earthdata login for unauthorized requests to /s3credentials endpoint.', async () => {
-      const response = await got(
-        `${process.env.DISTRIBUTION_ENDPOINT}/s3credentials`,
-        { followRedirect: false }
-      );
-      const authorizeUrl = new URL(response.headers.location);
-      expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
-      expect(authorizeUrl.searchParams.get('state')).toEqual('/s3credentials');
-      expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+      it('redirects to Earthdata login for requests on /s3credentials endpoint.', async () => {
+        const response = await got(
+          `${process.env.DISTRIBUTION_ENDPOINT}/s3credentials`,
+          { followRedirect: false }
+        );
+        const authorizeUrl = new URL(response.headers.location);
+        expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
+        expect(authorizeUrl.searchParams.get('state')).toEqual('/s3credentials');
+        expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+      });
+
+      it('downloads a public science file', async () => {
+        accessToken = randomId('accessToken');
+        const s3SignedUrl = await invokeLambdaForS3SignedUrl(publicFileUrl, accessToken);
+        const parts = new URL(s3SignedUrl);
+        const userName = parts.searchParams.get('x-EarthdataLoginUsername');
+
+        const fileStream = got.stream(s3SignedUrl);
+        const downloadChecksum = await generateChecksumFromStream('cksum', fileStream);
+        expect(userName).toEqual('publicAccess');
```

Comment:
Is this because every access on the distribution API needs a username for auditing?

Reply:
No, it's because I assumed that creating an s3 signature required it, but it looks like it might be optional.  So I could skip adding the `'x-EarthdataLoginUsername'` on the returned URL.

I'll ask Patrick if he cares.  And this could probably be removed.  

https://github.com/nasa/cumulus/blob/CUMULUS-1223-s3-public-access/packages/api/endpoints/distribution.js#L33

Reply:
Why would we remove that whole helper? We still want to set usernames for protected files

Reply:
discussion with @/bilts says to use  `unauthenticated` as the usernam.